### PR TITLE
Add optimization guide doc

### PR DESCRIPTION
**Pull request by sneaxiy**
As the title.

### CODE REVIEW

File: doc/tutorial/optimization_guide.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# Use SQLFlow to Solve Optimization Problems
+
+This document explains how to use the SQLFlow extended syntax to solve the optimization problems.
```
Comment (typhoonzero) [inline doc/tutorial/optimization_guide.md]: remove "extended syntax"?

File: doc/tutorial/optimization_guide.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+   product  amount
+   
+0  soldier      20
+
+1    train      60
+
+Saved in my_db.woodcarving_result_table.
+```
+
+We can also see the solved result by the SQL statement `SELECT * FROM my_db.woodcarving_result_table;`:
```
Comment (typhoonzero) [inline doc/tutorial/optimization_guide.md]: see -> examine

File: doc/tutorial/optimization_guide.md (hunk 1 above)
Comment (sneaxiy) [reply doc/tutorial/optimization_guide.md]: Done.

File: doc/tutorial/optimization_guide.md (hunk 2 above)
Comment (sneaxiy) [reply doc/tutorial/optimization_guide.md]: Done.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**